JavaDoc related build warnings

A `mvn clean package` call produces the following output:

```
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 1 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 1 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO] 
[INFO] -----------------------< ch.islandsql:islandsql >-----------------------
[INFO] Building islandsql 0.2.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ islandsql ---
[INFO] Deleting /Users/phs/github/IslandSQL/IslandSQL/target
[INFO] 
[INFO] --- antlr4-maven-plugin:4.11.1:antlr4 (default) @ islandsql ---
[INFO] ANTLR 4: Processing source directory /Users/phs/github/IslandSQL/IslandSQL/src/main/antlr4
[INFO] Processing grammar: ch/islandsql/grammar/IslandSqlLexer.g4
[INFO] Processing grammar: ch/islandsql/grammar/IslandSqlParser.g4
[INFO] 
[INFO] --- build-helper-maven-plugin:3.3.0:add-source (addSource) @ islandsql ---
[INFO] Source directory: /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/java added.
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ islandsql ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/phs/github/IslandSQL/IslandSQL/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.10.1:compile (default-compile) @ islandsql ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 9 source files to /Users/phs/github/IslandSQL/IslandSQL/target/classes
[INFO] 
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ islandsql ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/phs/github/IslandSQL/IslandSQL/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.10.1:testCompile (default-testCompile) @ islandsql ---
[INFO] No sources to compile
[INFO] 
[INFO] --- maven-surefire-plugin:2.12.4:test (default-test) @ islandsql ---
[INFO] No tests to run.
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ islandsql ---
[INFO] Building jar: /Users/phs/github/IslandSQL/IslandSQL/target/islandsql-0.2.0-SNAPSHOT.jar
[INFO] 
[INFO] --- maven-source-plugin:3.2.1:jar-no-fork (attach-sources) @ islandsql ---
[INFO] Building jar: /Users/phs/github/IslandSQL/IslandSQL/target/islandsql-0.2.0-SNAPSHOT-sources.jar
[INFO] 
[INFO] --- maven-javadoc-plugin:3.4.1:jar (attach-javadocs) @ islandsql ---
[INFO] No previous run data found, generating javadoc.
[WARNING] Javadoc Warnings
[WARNING] Loading source files for package ch.islandsql.grammar...
[WARNING] Constructing Javadoc information...
[WARNING] Building index for all the packages and classes...
[WARNING] Standard Doclet version 17.0.4.1+1-LTS-2
[WARNING] Building tree for all the packages and classes...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/FindRuleListener.html...
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/src/main/java/ch/islandsql/grammar/FindRuleListener.java:95: warning: no @param for <T>
[WARNING] public <T extends ParseTree> List<T> getResult() {
[WARNING] ^
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/IslandSqlDocument.html...
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/src/main/java/ch/islandsql/grammar/IslandSqlDocument.java:85: warning: no @param for <T>
[WARNING] public <T extends ParseTree> List<T> getAllContentsOfType(ParseTree parseTree, Class<T> desiredType) {
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/src/main/java/ch/islandsql/grammar/IslandSqlDocument.java:98: warning: no @param for <T>
[WARNING] public <T extends ParseTree> List<T> getAllContentsOfType(Class<T> desiredType) {
[WARNING] ^
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/IslandSqlLexer.html...
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlLexer.java:13: warning: no comment
[WARNING] public class IslandSqlLexer extends IslandSqlLexerBase {
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlLexer.java:416: warning: no comment
[WARNING] public static final ATN _ATN =
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlLexer.java:16: warning: no comment
[WARNING] protected static final DFA[] _decisionToDFA;
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlLexer.java:165: warning: no comment
[WARNING] public static final String _serializedATN =
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlLexer.java:17: warning: no comment
[WARNING] protected static final PredictionContextCache _sharedContextCache =
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlLexer.java:22: warning: no comment
[WARNING] ANY_OTHER=10;
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlLexer.java:23: warning: no comment
[WARNING] public static String[] channelNames = {
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlLexer.java:21: warning: no comment
[WARNING] CONDITIONAL_COMPILATION_DIRECTIVE=6, PLSQL_DECLARATION=7, SELECT=8, WS=9,
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlLexer.java:20: warning: no comment
[WARNING] ML_COMMENT=1, SL_COMMENT=2, REMARK_COMMAND=3, PROMPT_COMMAND=4, STRING=5,
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlLexer.java:27: warning: no comment
[WARNING] public static String[] modeNames = {
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlLexer.java:21: warning: no comment
[WARNING] CONDITIONAL_COMPILATION_DIRECTIVE=6, PLSQL_DECLARATION=7, SELECT=8, WS=9,
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlLexer.java:20: warning: no comment
[WARNING] ML_COMMENT=1, SL_COMMENT=2, REMARK_COMMAND=3, PROMPT_COMMAND=4, STRING=5,
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlLexer.java:20: warning: no comment
[WARNING] ML_COMMENT=1, SL_COMMENT=2, REMARK_COMMAND=3, PROMPT_COMMAND=4, STRING=5,
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlLexer.java:39: warning: no comment
[WARNING] public static final String[] ruleNames = makeRuleNames();
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlLexer.java:21: warning: no comment
[WARNING] CONDITIONAL_COMPILATION_DIRECTIVE=6, PLSQL_DECLARATION=7, SELECT=8, WS=9,
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlLexer.java:20: warning: no comment
[WARNING] ML_COMMENT=1, SL_COMMENT=2, REMARK_COMMAND=3, PROMPT_COMMAND=4, STRING=5,
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlLexer.java:20: warning: no comment
[WARNING] ML_COMMENT=1, SL_COMMENT=2, REMARK_COMMAND=3, PROMPT_COMMAND=4, STRING=5,
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlLexer.java:54: warning: no comment
[WARNING] public static final Vocabulary VOCABULARY = new VocabularyImpl(_LITERAL_NAMES, _SYMBOLIC_NAMES);
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlLexer.java:21: warning: no comment
[WARNING] CONDITIONAL_COMPILATION_DIRECTIVE=6, PLSQL_DECLARATION=7, SELECT=8, WS=9,
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlLexer.java:88: warning: no comment
[WARNING] public IslandSqlLexer(CharStream input) {
[WARNING] ^
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/IslandSqlLexerBase.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/IslandSqlParser.html...
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:13: warning: no comment
[WARNING] public class IslandSqlParser extends Parser {
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:96: warning: no comment
[WARNING] public static class FileContext extends ParserRuleContext {
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:160: warning: no comment
[WARNING] public static class SelectStatementContext extends ParserRuleContext {
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:223: warning: no comment
[WARNING] public static final ATN _ATN =
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:16: warning: no comment
[WARNING] protected static final DFA[] _decisionToDFA;
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:212: warning: no comment
[WARNING] public static final String _serializedATN =
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:17: warning: no comment
[WARNING] protected static final PredictionContextCache _sharedContextCache =
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:22: warning: no comment
[WARNING] ANY_OTHER=10;
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:21: warning: no comment
[WARNING] CONDITIONAL_COMPILATION_DIRECTIVE=6, PLSQL_DECLARATION=7, SELECT=8, WS=9,
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:20: warning: no comment
[WARNING] ML_COMMENT=1, SL_COMMENT=2, REMARK_COMMAND=3, PROMPT_COMMAND=4, STRING=5,
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:21: warning: no comment
[WARNING] CONDITIONAL_COMPILATION_DIRECTIVE=6, PLSQL_DECLARATION=7, SELECT=8, WS=9,
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:20: warning: no comment
[WARNING] ML_COMMENT=1, SL_COMMENT=2, REMARK_COMMAND=3, PROMPT_COMMAND=4, STRING=5,
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:20: warning: no comment
[WARNING] ML_COMMENT=1, SL_COMMENT=2, REMARK_COMMAND=3, PROMPT_COMMAND=4, STRING=5,
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:24: warning: no comment
[WARNING] RULE_file = 0, RULE_selectStatement = 1;
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:24: warning: no comment
[WARNING] RULE_file = 0, RULE_selectStatement = 1;
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:30: warning: no comment
[WARNING] public static final String[] ruleNames = makeRuleNames();
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:21: warning: no comment
[WARNING] CONDITIONAL_COMPILATION_DIRECTIVE=6, PLSQL_DECLARATION=7, SELECT=8, WS=9,
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:20: warning: no comment
[WARNING] ML_COMMENT=1, SL_COMMENT=2, REMARK_COMMAND=3, PROMPT_COMMAND=4, STRING=5,
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:20: warning: no comment
[WARNING] ML_COMMENT=1, SL_COMMENT=2, REMARK_COMMAND=3, PROMPT_COMMAND=4, STRING=5,
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:45: warning: no comment
[WARNING] public static final Vocabulary VOCABULARY = new VocabularyImpl(_LITERAL_NAMES, _SYMBOLIC_NAMES);
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:21: warning: no comment
[WARNING] CONDITIONAL_COMPILATION_DIRECTIVE=6, PLSQL_DECLARATION=7, SELECT=8, WS=9,
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:90: warning: no comment
[WARNING] public IslandSqlParser(TokenStream input) {
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:123: warning: no comment
[WARNING] public final FileContext file() throws RecognitionException {
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:182: warning: no comment
[WARNING] public final SelectStatementContext selectStatement() throws RecognitionException {
[WARNING] ^
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/IslandSqlParser.FileContext.html...
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:104: warning: no comment
[WARNING] public FileContext(ParserRuleContext parent, int invokingState) {
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:97: warning: no comment
[WARNING] public TerminalNode EOF() { return getToken(IslandSqlParser.EOF, 0); }
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:98: warning: no comment
[WARNING] public List<SelectStatementContext> selectStatement() {
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:101: warning: no comment
[WARNING] public SelectStatementContext selectStatement(int i) {
[WARNING] ^
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/IslandSqlParser.SelectStatementContext.html...
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:163: warning: no comment
[WARNING] public SelectStatementContext(ParserRuleContext parent, int invokingState) {
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:161: warning: no comment
[WARNING] public TerminalNode PLSQL_DECLARATION() { return getToken(IslandSqlParser.PLSQL_DECLARATION, 0); }
[WARNING] ^
[WARNING] /Users/phs/github/IslandSQL/IslandSQL/target/generated-sources/antlr4/ch/islandsql/grammar/IslandSqlParser.java:162: warning: no comment
[WARNING] public TerminalNode SELECT() { return getToken(IslandSqlParser.SELECT, 0); }
[WARNING] ^
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/IslandSqlParserBaseListener.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/IslandSqlParserBaseVisitor.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/IslandSqlParserListener.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/IslandSqlParserVisitor.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/package-summary.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/package-tree.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/constant-values.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/class-use/IslandSqlLexerBase.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/class-use/FindRuleListener.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/class-use/IslandSqlDocument.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/class-use/IslandSqlParser.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/class-use/IslandSqlParser.SelectStatementContext.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/class-use/IslandSqlParser.FileContext.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/class-use/IslandSqlParserVisitor.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/class-use/IslandSqlLexer.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/class-use/IslandSqlParserListener.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/class-use/IslandSqlParserBaseVisitor.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/class-use/IslandSqlParserBaseListener.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/ch/islandsql/grammar/package-use.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/overview-tree.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/deprecated-list.html...
[WARNING] Building index for all classes...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/allclasses-index.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/allpackages-index.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/index-all.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/index.html...
[WARNING] Generating /Users/phs/github/IslandSQL/IslandSQL/target/apidocs/help-doc.html...
[WARNING] 54 warnings
[INFO] Building jar: /Users/phs/github/IslandSQL/IslandSQL/target/islandsql-0.2.0-SNAPSHOT-javadoc.jar
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.456 s
[INFO] Finished at: 2023-02-03T12:08:08+01:00
[INFO] ------------------------------------------------------------------------
```

The JavaDoc related warnings should be fixed.

--- a/pom.xml
+++ b/pom.xml
@@ -105,6 +105,9 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-javadoc-plugin</artifactId>
                 <version>3.4.1</version>
+                <configuration>
+                    <doclint>all,-missing</doclint>
+                </configuration>
                 <executions>
                     <execution>
                         <id>attach-javadocs</id>

--- a/src/main/java/ch/islandsql/grammar/FindRuleListener.java
+++ b/src/main/java/ch/islandsql/grammar/FindRuleListener.java
@@ -89,7 +89,7 @@ public class FindRuleListener implements ParseTreeListener {
     /**
      * Return the result after walking the parse-tree.
      *
-     * @return list of nodes matching the desired class.
+     * @return List of nodes matching the desired class.
      * @param <T> The return type of the result.
      */
     @SuppressWarnings("unchecked")

--- a/src/main/java/ch/islandsql/grammar/FindRuleListener.java
+++ b/src/main/java/ch/islandsql/grammar/FindRuleListener.java
@@ -90,6 +90,7 @@ public class FindRuleListener implements ParseTreeListener {
      * Return the result after walking the parse-tree.
      *
      * @return list of nodes matching the desired class.
+     * @param <T> The return type of the result.
      */
     @SuppressWarnings("unchecked")
     public <T extends ParseTree> List<T> getResult() {

--- a/src/main/java/ch/islandsql/grammar/IslandSqlDocument.java
+++ b/src/main/java/ch/islandsql/grammar/IslandSqlDocument.java
@@ -81,6 +81,7 @@ public class IslandSqlDocument {
      * @param parseTree start node.
      * @param desiredType desired class (must be a descendant of ParseTree).
      * @return list of nodes that are instances of the of desired class.
+     * @param <T> The return type of the result.
      */
     public <T extends ParseTree> List<T> getAllContentsOfType(ParseTree parseTree, Class<T> desiredType) {
         FindRuleListener listener = new FindRuleListener(desiredType);
@@ -94,6 +95,7 @@ public class IslandSqlDocument {
      *
      * @param desiredType desired class (must be a descendant of ParseTree).
      * @return list of nodes that are instances of the of desired class.
+     * @param <T> The return type of the result.
      */
     public <T extends ParseTree> List<T> getAllContentsOfType(Class<T> desiredType) {
         return getAllContentsOfType(file, desiredType);

--- a/src/main/java/ch/islandsql/grammar/IslandSqlDocument.java
+++ b/src/main/java/ch/islandsql/grammar/IslandSqlDocument.java
@@ -49,7 +49,7 @@ public class IslandSqlDocument {
      * Factory to construct an IslandSqlDocument.
      *
      * @param sql SQL-script as string.
-     * @return constructed IslandSqlDocument,
+     * @return Constructed IslandSqlDocument.
      */
     public static IslandSqlDocument parse(String sql) {
         return new IslandSqlDocument(sql);
@@ -59,7 +59,7 @@ public class IslandSqlDocument {
      * Returns the token stream produced by the lexer.
      * Can be used to access hidden tokens.
      *
-     * @return token stream.
+     * @return Token stream.
      */
     public CommonTokenStream getTokenStream() {
         return tokenStream;
@@ -68,7 +68,7 @@ public class IslandSqlDocument {
     /**
      * Returns the start node of the parse tree.
      *
-     * @return file (start rule).
+     * @return File (start rule).
      */
     public IslandSqlParser.FileContext getFile() {
         return file;
@@ -78,9 +78,9 @@ public class IslandSqlDocument {
      * Gets all nodes that are instances of a desired class.
      * Start node can be any node in the parse tree.
      *
-     * @param parseTree start node.
-     * @param desiredType desired class (must be a descendant of ParseTree).
-     * @return list of nodes that are instances of the of desired class.
+     * @param parseTree Start node.
+     * @param desiredType Desired class (must be a descendant of ParseTree).
+     * @return List of nodes that are instances of the of desired class.
      * @param <T> The return type of the result.
      */
     public <T extends ParseTree> List<T> getAllContentsOfType(ParseTree parseTree, Class<T> desiredType) {
@@ -89,12 +89,13 @@ public class IslandSqlDocument {
         walker.walk(listener, parseTree);
         return listener.getResult();
     }
+
     /**
      * Gets all nodes that are instances of a desired class.
      * Start node is file.
      *
-     * @param desiredType desired class (must be a descendant of ParseTree).
-     * @return list of nodes that are instances of the of desired class.
+     * @param desiredType Desired class (must be a descendant of ParseTree).
+     * @return List of nodes that are instances of the of desired class.
      * @param <T> The return type of the result.
      */
     public <T extends ParseTree> List<T> getAllContentsOfType(Class<T> desiredType) {

--- a/src/main/java/ch/islandsql/grammar/IslandSqlLexerBase.java
+++ b/src/main/java/ch/islandsql/grammar/IslandSqlLexerBase.java
@@ -30,7 +30,7 @@ public abstract class IslandSqlLexerBase extends Lexer {
     /**
      * Constructor.
      *
-     * @param input a character stream
+     * @param input A character stream.
      */
     public IslandSqlLexerBase(CharStream input) {
         super(input);
@@ -41,7 +41,7 @@ public abstract class IslandSqlLexerBase extends Lexer {
      * Must be implemented as function returning a boolean value to ensure
      * it is executed.
      *
-     * @return true (always).
+     * @return Returns always true.
      */
     @SuppressWarnings("SameReturnValue")
     public boolean saveQuoteDelimiter1() {
@@ -53,7 +53,7 @@ public abstract class IslandSqlLexerBase extends Lexer {
      * Determines if the character two positions before the current character
      * is equal to the character saved in "quoteDelimiter1".
      *
-     * @return true if character matches quoteDelimiter1.
+     * @return Returns true if character matches quoteDelimiter1.
      */
     public boolean checkQuoteDelimiter2() {
         String quoteDelimiter2 = _input.getText(new Interval(_input.index()-2, _input.index()-2));
@@ -64,9 +64,9 @@ public abstract class IslandSqlLexerBase extends Lexer {
      * Determines if the character at the passed position is a valid character.
      * Begin-of-file is considered a valid character.
      *
-     * @param validChars string containing valid characters.
-     * @param pos the position in the character stream.
-     * @return true if the character at the passed position is a valid character.
+     * @param validChars A string containing valid characters.
+     * @param pos The position in the character stream.
+     * @return Returns true if the character at the passed position is a valid character.
      */
     private boolean isCharOneOf(String validChars, int pos) {
         if (pos < 0) {
@@ -83,7 +83,7 @@ public abstract class IslandSqlLexerBase extends Lexer {
      * allowed. In other words a command can start in any column.
      * A command can start at begin-of-file.
      *
-     * @return true if the current position is valid for a command.
+     * @return Returns true if the current position is valid for a command.
      */
     public boolean isBeginOfCommand() {
         int i = _input.index() - 1;
@@ -103,7 +103,7 @@ public abstract class IslandSqlLexerBase extends Lexer {
      * single line are allowed.
      * A SQL statement can start at begin-of-file.
      *
-     * @return true if the current position is valid for a SQL statement.
+     * @return Returns true if the current position is valid for a SQL statement.
      */
     public boolean isBeginOfStatement() {
         return isCharOneOf(" \t\r\n;", _input.index()-1);

